core/src/main/scala/dagr/core/tasksystem/Pipeline.scala

```diff
@@ -59,6 +62,17 @@ abstract class Pipeline(val outputDirectory: Option[Path] = None) extends Task w
   final override def getTasks: Traversable[_ <: Task] = {
     build()
     tasks.toList.foreach(addChildren)
+
+    // Do the necessary prefixing and suffixing of task names
+    if (prefix.isDefined || suffix.isDefined) {
+      tasks.foreach {
+        case p: Pipeline =>
+          p.prefix = if (p.prefix.isEmpty) prefix else p.prefix.map(prefix.getOrElse("") + _)
+          p.suffix = if (p.suffix.isEmpty) suffix else p.suffix.map(suffix.getOrElse("") + _)
+        case t: Task     => t.name = prefix.getOrElse("") + t.name + suffix.getOrElse("")
```

Comment:
I don't think it is that big of a concern now and more of a caution, but the list of types of tasks that may need to be matched here could grow based on naming of tasks names.